### PR TITLE
Update Winapp3.ini

### PR DESCRIPTION
**Pull request by decofona**
Now corrected everything.

### CODE REVIEW

**Comment by APMichael** [comment]
Thanks again, but unfortunately there are still some syntax errors. All ExcludeKeys must have the "\\|" even if no wildcards are used:

**Correct:**
ExcludeKey1=FILE|%LocalAppData%\\Microsoft\\Windows\\INetCache\\IE\\|container.dat

**Wrong:**
ExcludeKey1=FILE|%LocalAppData%\\Microsoft\\Windows\\INetCache\\IE\\container.dat
ExcludeKey1=FILE|%LocalAppData%\\Microsoft\\Windows\\INetCache\\IE|container.dat

File: Winapp3/Winapp3.ini [hunk 1]
```diff
@@ -261,6 +261,44 @@ ExcludeKey66=PATH|%WinDir%\SysWOW64\config\|SOFTWARE.bak*
...
+ExcludeKey70=PATH|%WinDir%\SysWOW64\config\systemprofile\AppData\Local\|*.tmp
+ExcludeKey71=PATH|%WinDir%\SysWOW64\config\systemprofile\AppData\Local\DataSharing\Storage\|*.chk
+ExcludeKey72=PATH|%WinDir%\SysWOW64\config\systemprofile\AppData\Local\Microsoft\Windows\WebCache\|*.chk
+ExcludeKey73=PATH|%WinDir%\Panther\|*.tmp
+ExcludeKey74=PATH|%ProgramFiles%\Adobe\Adobe Media Encoder CC 2019|*.bak
+ExcludeKey75=PATH|%LocalAppData%\Packages\Microsoft.Windows.Cortana_cw5n1h2txyewy\AC\AppCache\EGFPDUEB\1\|*.txt
+ExcludeKey76=FILE|%ProgramFiles%\Adobe\Adobe Photoshop CC 2019\Required\Linguistics\Providers\Plugins2\AdobeHunspellPlugin\dictionaries\en_GB\changelog.txt
+ExcludeKey77=FILE|%ProgramFiles%\Adobe\Adobe Photoshop CC 2019\Required\Linguistics\Providers\Plugins2\AdobeHunspellPlugin\dictionaries\es_ES\Changelog.txt
+ExcludeKey78=FILE|%ProgramFiles%\Adobe\Adobe Photoshop CC 2019\Required\Linguistics\Providers\Plugins2\AdobeHunspellPlugin\dictionaries\lv_LV\Changelog.txt
+ExcludeKey79=FILE|%ProgramFiles%\Adobe\Adobe Photoshop CC 2019\Required\Linguistics\Providers\Plugins2\AdobeHunspellPlugin\dictionaries\th_th\ChangeLog.txt
```
Comment (APMichael) [inline Winapp3/Winapp3.ini]: For ExcludeKey76-79 you can use:
"%ProgramFiles%\\Adobe\\Adobe Photoshop CC \*\\Required\\Linguistics\\Providers\\Plugins2\\AdobeHunspellPlugin\\dictionaries\\\*\\|changelog.txt"

File: Winapp3/Winapp3.ini [hunk 2]
```diff
@@ -261,6 +261,44 @@ ExcludeKey66=PATH|%WinDir%\SysWOW64\config\|SOFTWARE.bak*
...
+ExcludeKey80=FILE|%ProgramFiles%\Adobe\Adobe Premiere Pro CC 2019\|*.bak
+ExcludeKey81=FILE|%ProgramFiles%\ImageMagick-7.0.8-Q16\ChangeLog.txt
+ExcludeKey82=FILE|%ProgramFiles%\ImageMagick-7.0.8-Q16\www\api\MagickCore\struct__ErrorInfo.html
+ExcludeKey83=FILE|%ProgramFiles%\LibreOffice\share\config\soffice.cfg\modules\dbapp\toolbar\reportobjectbar.xml
+ExcludeKey84=FILE|%ProgramFiles%\LibreOffice\share\config\soffice.cfg\modules\dbreport\popupmenu\report.xml
+ExcludeKey85=FILE|%ProgramFiles%\LibreOffice\share\config\soffice.cfg\modules\dbreport\toolbar\reportcontrols.xml
+ExcludeKey86=FILE|%ProgramFiles%\LibreOffice\share\extensions\dict-en\changelog.txt
+ExcludeKey87=FILE|%ProgramFiles%\LibreOffice\share\extensions\dict-it\CHANGELOG.txt
```
Comment (APMichael) [inline Winapp3/Winapp3.ini]: For ExcludeKey86+87 you can use:
"%ProgramFiles%\\LibreOffice\\share\\extensions\\\*\\|changelog.txt"

File: Winapp3/Winapp3.ini [hunk 3]
```diff
@@ -261,6 +261,44 @@ ExcludeKey66=PATH|%WinDir%\SysWOW64\config\|SOFTWARE.bak*
...
+ExcludeKey88=FILE|%ProgramFiles%\Microsoft Office\Office16\SLERROR.XML
+ExcludeKey89=FILE|%ProgramFiles%\FormatFactory\updatelog.txt
+ExcludeKey90=PATH|%WinDir%\appcompat\Programs\|*.LOG;*.LOG*
```
Comment (APMichael) [inline Winapp3/Winapp3.ini]: For "*.LOG;*.LOG*" you can just use "*.LOG*" (ExcludeKey90+91+93).

And I have to repeat: Please always use Winapp2ool > WinappDebug to make sure the syntax and sorting is correct before opening a new pull request. Thank you very much.

**Comment by decofona** [comment]
Okay I will try it now with Winapp2ool and come back to you.

**Comment by decofona** [comment]
I actually used, it has found 38 possible errors in Winapp3.ini. Do I have to manually correct and sort them or this will be done automatically?

**Comment by APMichael** [comment]
You can enable saving for this. (Please note: if you don't change the filename, the file will be overwritten.)

**Comment by APMichael** [comment]
Thank you. At first glance, it looks good now. I will review the changes tomorrow.

**Comment by decofona** [comment]
Okay, thanks.

**Comment by APMichael** [comment]
I made some corrections for you:
- Replaced %AppData% and %LocalAppData% to make the ExcludeKeys multi-user compatible.
- Replaced program version numbers with wildcards.
- Removed unnecessary "All Users" entries, because "All Users" is only a junction to %CommonAppData%.

**Comment by decofona** [comment]
Yeah you can merge it.

**Comment by APMichael** [comment]
Thank you for your reply. I will merge the pull request in the next few days.

**Comment by decofona** [comment]
Hello, while testing out the entry UselessFiles a slight error has been found by me:

C:\Users\All Users\Microsoft\SmsRouter\MessageStore\edb.chk	8 KB

C:\Users\{Username}\AppData\Local\Microsoft\Windows \Explorer\ThumbCacheToDelete\thm99DA.tmp	1.024 KB

I think the All Users entry we have to add like

%SystemDrive%\All Users\Microsoft\SmsRouter\MessageStore\|*chk

And the other one like this:

%SystemDrive%\*\AppData\Local\Microsoft\Windows \Explorer\ThumbCacheToDelete\|*.tmp

**Comment by APMichael** [comment]
Okay, unfortunately, it seems that CCleaner doesn't realize that "C:\\Users\\All Users\\" is "%CommonAppData%". I added the necessary ExcludeKeys again and optimized the ExcludeKey for "\*CacheToDelete".

**Comment by decofona** [comment]
I tested out the entry and it works perfectly as it should. Thanks a lot!

**Pull request merged**